src/directive/validators.ts

```diff
@@ -0,0 +1,28 @@
+import Joi from "joi";
+import type { CreateDirectiveRequest } from "./model";
+
+export const validateCreateDirectiveRequest = (
+  data: unknown
+): data is CreateDirectiveRequest => {
+  const validation = Joi.object()
+    .keys({
+      archiveId: Joi.number().integer().min(1).required(),
+      stewardAccountId: Joi.when("type", {
+        is: Joi.string().valid("transfer"),
+        then: Joi.number().integer().min(1).required(),
+        otherwise: Joi.valid(null),
+      }),
+      type: Joi.string().required(),
+      note: Joi.string(),
+      trigger: Joi.object()
+        .keys({
+          type: Joi.string().required(),
+        })
+        .required(),
+    })
```

Comment:
So my request failed until I added `emailFromAuthToken: Joi.string().email(),` in the validation rules. Without that, I got:

```
[Error [ValidationError]: "emailFromAuthToken" is not allowed] {
  _original: {
    archiveId: 1,
    stewardAccountId: 3,
    type: 'transfer',
    trigger: { type: 'admin' },
    emailFromAuthToken: 'cecilia+local@permanent.org'
  },
  details: [
    {
      message: '"emailFromAuthToken" is not allowed',
      path: [Array],
      type: 'object.unknown',
      context: [Object]
    }
  ]
}
```

With that new rule, I could send a request without an email and have it succeed, which is what I expected (since I see you're inserting the email from the auth token separately, as only makes sense). I guess Joi limits the body to only the fields in the validator. Presumably you didn't run into this though, so I don't know why it appeared for me!  Maybe there's something else going on?

Reply:
I did run into it, but I apparently fixed it on the wrong branch! Will fix here too.